src/Http/Http.Results/tools/ResultsOfTGenerator/Program.cs

```diff
@@ -166,6 +173,17 @@ static void GenerateClassFile(string classFilePath, int typeArgCount, bool inter
                     writer.WriteLine();
                 }
             }
+            writer.WriteLine();
+
+            // IEndpointMetadataProvider.PopulateMetadata
+            writer.WriteIndentedLine("/// <inheritdoc/>");
+            writer.WriteIndentedLine("public static void PopulateMetadata(EndpointMetadataContext context)");
```

Comment:
```suggestion
            writer.WriteIndentedLine("static void IEndpointMetadataProvider.PopulateMetadata(EndpointMetadataContext context)");
```

Do we expect people to be calling this without the interface? `Results<MyType1, MyType2>.PopulateMetadata(context);` doesn't seem super useful. We could always make fully public later.

Reply:
Looks like you can.

Reply:
<img width="866" alt="image" src="https://user-images.githubusercontent.com/249088/162498432-07b2899a-95d8-4f8b-9687-b4f3bcd732fb.png">

Reply:
> Can you do explicit static abstract interface implementations?

Yeah. A bunch of generic math interfaces were just mad implicit instead of explicit, but I think implicit implementation makes a lot more sense for methods like `TryParse`. https://github.com/dotnet/runtime/pull/66748